tools/cdk-build-tools/lib/compile.ts

```diff
@@ -1,14 +1,25 @@
 import { makeExecutable, shell } from './os';
-import { CDKBuildOptions, CompilerOverrides, currentPackageJson, packageCompiler } from './package-info';
+import { CDKBuildOptions, CompilerOverrides, currentPackageJson, isJsii, packageCompiler } from './package-info';
 import { Timers } from './timer';
 
+const DEFAULT_ROSETTA = require.resolve('jsii-rosetta/bin/jsii-rosetta');
+
 /**
  * Run the compiler on the current package
  */
 export async function compileCurrentPackage(options: CDKBuildOptions, timers: Timers, compilers: CompilerOverrides = {}): Promise<void> {
   const env = options.env;
   await shell(packageCompiler(compilers), { timers, env });
 
+  if (isJsii() && options.rosetta === 'strict') {
```

Comment:
I'd rather we don't do this here and make it a separate npm `script` for developers to run, OR only do this conditionally under an environment variable.

This effectively runs compilation twice, and we're already none too happy about our build times.

Alternatively, add a way to rosetta to attach metadata to collected snippets to say "if this individual snippet fails to compile, then fail" (as opposed to that having be a flag to the entire operation).